service/src/main/java/bio/terra/pipelines/service/ImputationService.java

```diff
@@ -1,40 +1,140 @@
 import bio.terra.pipelines.dependencies.leonardo.LeonardoService;
 import bio.terra.pipelines.dependencies.leonardo.LeonardoServiceException;
 import bio.terra.pipelines.dependencies.sam.SamService;
+import bio.terra.pipelines.dependencies.stairway.JobBuilder;
+import bio.terra.pipelines.dependencies.stairway.JobMapKeys;
+import bio.terra.pipelines.dependencies.stairway.JobService;
 import bio.terra.pipelines.dependencies.wds.WdsService;
 import bio.terra.pipelines.dependencies.wds.WdsServiceException;
+import bio.terra.pipelines.stairway.RunImputationJobFlight;
+import bio.terra.pipelines.stairway.RunImputationJobFlightMapKeys;
+import com.google.common.annotations.VisibleForTesting;
 import java.util.Collections;
 import java.util.List;
+import java.util.UUID;
 import org.broadinstitute.dsde.workbench.client.leonardo.model.ListAppResponse;
+import org.hibernate.exception.ConstraintViolationException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataIntegrityViolationException;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
 /** Service to encapsulate logic used to run an imputation pipeline */
 @Service
 public class ImputationService {
   private static final Logger logger = LoggerFactory.getLogger(ImputationService.class);
+  private final ImputationJobsRepository imputationJobsRepository;
+  private final PipelineInputsRepository pipelineInputsRepository;
   private LeonardoService leonardoService;
   private SamService samService;
   private WdsService wdsService;
+  private final JobService jobService;
   private ImputationConfiguration imputationConfiguration;
 
   @Autowired
   ImputationService(
+      ImputationJobsRepository imputationJobsRepository,
+      PipelineInputsRepository pipelineInputsRepository,
       LeonardoService leonardoService,
       SamService samService,
       WdsService wdsService,
+      JobService jobService,
       ImputationConfiguration imputationConfiguration) {
+    this.imputationJobsRepository = imputationJobsRepository;
+    this.pipelineInputsRepository = pipelineInputsRepository;
     this.leonardoService = leonardoService;
     this.samService = samService;
     this.wdsService = wdsService;
+    this.jobService = jobService;
     this.imputationConfiguration = imputationConfiguration;
   }
 
+  /**
+   * Creates a new Imputation pipeline service job, using a Stairway flight, based on a user's
+   * request. Returns jobId of new job (which is the same as the flightId) if flight submission is
+   * successful.
+   *
+   * @param userId
+   * @param pipelineVersion
+   * @return String jobId
+   *     <p>Note that the information in the requested job will grow over time, along with the
+   *     following related classes:
+   * @see ImputationJob
+   */
+  @Transactional
```

Comment:
does this need to be transactional.  I'm assuming this is just doing one write to the stairway database

Reply:
hmm at this time that's true. in future we might be doing more that might merit its being transactional, but happy to take this out for now